src/NuGet.Core/NuGet.DependencyResolver.Core/Remote/RemoteWalkContext.cs

```diff
@@ -46,7 +40,9 @@ public RemoteWalkContext(SourceCacheContext cacheContext, PackageSourceMapping p
         /// <summary>
         /// Library entry cache.
         /// </summary>
-        public ConcurrentDictionary<LibraryRangeCacheKey, Task<GraphItem<RemoteResolveResult>>> FindLibraryEntryCache { get; }
+        internal static TaskResultCache<LibraryRangeCacheKey, GraphItem<RemoteResolveResult>> FindLibraryEntryCache { get; } = new();
```

Comment:
I don't think this is the right thing. This keeps all the tasks forever. 

This change has nothing to do with the continuation.

Reply:
another notable thing is that these caches are not necessarily save to be shared across projects as is. 

GraphItem has concepts like CentralTransitive which can only ever be project specific. 

Does restore with this change yield the same results?

Reply:
While I like the enthusiam here, I agree with @nkolev92 it doesn't feel like GraphItem is at the right level to be caching here.

Reply:
Sorry this was a mistake and was just me experimenting with all projects sharing the same task cache.  I've reverted this